File: codigo/backend/avaliai/src/main/java/com/ti/avaliai/subject/Subject.java
package com.ti.avaliai.subject;

import com.ti.avaliai.course.Course;
import com.ti.avaliai.review.Review;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

import static com.ti.avaliai.utils.HashUtils.generateHash;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "t_subject")
public class Subject {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name="id")
    private long id;

    @Column(name="hash_id")
    private String hashId = generateHash();

    @Column(name = "name")
    private String name;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name ="campus")
    private String campus;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "fk_course")
    private Course course;

    @Column(name = "short_description", columnDefinition = "TEXT")
    private String shortDescription;

    @Column(name = "long_description", columnDefinition = "TEXT")
    private String longDescription;

    @OneToMany(mappedBy = "subject", fetch = FetchType.EAGER)
    private List<Review> reviews;
}

File: codigo/front_mobile/lib/src/_subject.dart
class Subject {
  final int id;
  final String hashId;
  final String name;
  final String campus;
  final String course;
  final String university;
  final String courseHashId;
  final String universityHashId;
  final String shortDescription;
  final String longDescription;
  final double score;

  const Subject({required this.id, required this.hashId, required this.name, required this.campus, required this.course, required this.university, required this.courseHashId, required this.universityHashId, required this.shortDescription, required this.longDescription, required this.score});

  factory Subject.fromJson(Map<String, dynamic> json) {
    return Subject(id: json['id'], hashId: json['hashId'], name: json['name'], campus: json['campus'],course: json['course'], university: json['university'], courseHashId: json['courseHashId'],universityHashId: json['universityHashId'], shortDescription: json['shortDescription'], longDescription: json['longDescription'], score: json['score']);
  }
}